### DISCUSSION TITLE
Reaction control thruster actuation of a spacecraft model

### DISCUSSION THREAD

**Discussion by handsomir** [google-deepmind/mujoco#1300] [Show and tell]
I am learning MuJoCo while developing a dynamic model of a spacecraft for a space robotics project. So far, I have tested a model attitude (rotational) and translation control with thrusters in an environment without external perturbations. See the screenshot. Also, attitude control with reaction wheels works as expected. Next step is to simulate flexible appendages, e.g., long and skinny solar panels, and their deployment. 
![screenshot](https://github.com/google-deepmind/mujoco/assets/83923198/acbd02c0-8b0e-4fcf-9abe-977a7bddf0db)

**Comment by yuvaltassa** [comment]
For better integration of rotations, make sure to use the implicit integrator (or possibly RK4), as documented [here](https://mujoco.readthedocs.io/en/latest/computation/index.html#numerical-integration).

**Reply by handsomir** [reply]
Thank you @yuvaltassa. RK4 is my go-to integrator so I have selected it already for all my sims.

**Comment by tmplt** [comment]
Any chance you could share the model?

**Reply by handsomir** [reply]
@tmplt I have asked my superiors and been told that, albeit the model is a basic configuration, I am not allowed to share it. Things might change as we make progress with our projects. When I'm allowed to share I'll post in show and tell to help the community.

**Reply by tmplt** [reply]
I know the seat you're in. Nevertheless, thanks for the status update.

**Comment by Eren-01** [comment]
Could you please share the model? Thank you!
